### ISSUE TITLE
25 rue Louis de Broglie - ZAC de   la Duranne 13100 Aix-en-Provence

### ISSUE THREAD

**Issue by cquest** [addok/addok#215]
25 rue Louis de Broglie - ZAC de   la Duranne 13100 Aix-en-Provence
    Pas de résultat

C'est la mention de la ZAC qui pose problème...

EXPLAIN 25 rue Louis de Broglie 13100 Aix-en-Provence
[1.860] Taken tokens: [<Token brogli>, <Token duran>, <Token 13100>, <Token zak>, <Token aix>]
[1.888] Common tokens: [<Token loui>, <Token en>, <Token provens>, <Token 25>, <Token la>, <Token ru>, <Token de>, <Token de>]
[1.898] Not found tokens: []
[1.908] Filters: []
[1.919] ** STEP_ONLY_COMMONS **
[1.930] ** STEP_NO_MEANINGFUL_BUT_COMMON_TRY_AUTOCOMPLETE **
[1.939] ** STEP_BUCKET_WITH_MEANINGFUL **
[1.955] New bucket with keys ['w|brogli', 'w|duran', 'w|13100', 'w|zak', 'w|aix'] and limit 10
[2.214] 0 ids in bucket so far
[2.228] ** STEP_REDUCE_WITH_OTHER_COMMONS **
[2.243] ** STEP_ENSURE_GEOHASH_RESULTS_ARE_INCLUDED_IF_CENTER_IS_GIVEN **
[2.252] ** STEP_AUTOCOMPLETE **
[2.264] Autocompleting provens
[2.356] Found tokens to autocomplete set()
[2.370] ** STEP_CHECK_BUCKET_FULL **
[2.380] ** STEP_CHECK_CREAM **
[2.390] ** STEP_FUZZY **
[2.406] Fuzzy on. Trying with [<Token brogli>, <Token duran>, <Token 13100>, <Token zak>, <Token aix>].
[2.430] Going fuzzy with brogli
[4.196] Found fuzzy candidates []
[4.215] Going fuzzy with duran
[5.731] Found fuzzy candidates []
[5.752] Going fuzzy with zak
[6.784] Found fuzzy candidates []
[6.803] Going fuzzy with aix
[7.808] Found fuzzy candidates []
[7.831] Fuzzy on. Trying with [<Token brogli>, <Token duran>, <Token 13100>, <Token zak>, <Token aix>].
[7.849] Going fuzzy with brogli
[9.510] Found fuzzy candidates []
[9.529] Going fuzzy with duran
[10.98] Found fuzzy candidates []
[11.00] Going fuzzy with zak
[11.96] Found fuzzy candidates []
[11.97] Going fuzzy with aix
[12.93] Found fuzzy candidates []
[12.95] ** STEP_EXTEND_RESULTS_REDUCING_TOKENS **
[12.96] Bucket dry. Trying to remove some tokens.
[12.98] Adding to bucket with keys ['w|brogli', 'w|duran', 'w|zak', 'w|aix']
[13.18] 0 ids in bucket so far
[13.20] Adding to bucket with keys ['w|brogli', 'w|duran', 'w|13100', 'w|zak']
[13.38] 0 ids in bucket so far
[13.40] Adding to bucket with keys ['w|brogli', 'w|13100', 'w|zak', 'w|aix']
[13.58] 0 ids in bucket so far
[13.60] Adding to bucket with keys ['w|brogli', 'w|duran', 'w|13100', 'w|aix']
[13.84] 0 ids in bucket so far
[13.86] Adding to bucket with keys ['w|duran', 'w|13100', 'w|zak', 'w|aix']
[14.40] 0 ids in bucket so far
[14.41] Computing results
[14.42] Done computing results
14.4 ms / 0 results

> EXPLAIN 25 rue Louis de Broglie 13100 Aix-en-Provence
[1.269] Taken tokens: [<Token brogli>, <Token 13100>, <Token aix>]
[1.294] Common tokens: [<Token loui>, <Token en>, <Token provens>, <Token 25>, <Token ru>, <Token de>]
[1.304] Not found tokens: []
[1.315] Filters: []
[1.327] ** STEP_ONLY_COMMONS **
[1.337] ** STEP_NO_MEANINGFUL_BUT_COMMON_TRY_AUTOCOMPLETE **
[1.347] ** STEP_BUCKET_WITH_MEANINGFUL **
[1.362] New bucket with keys ['w|brogli', 'w|13100', 'w|aix'] and limit 10
[1.662] 0 ids in bucket so far
[1.693] ** STEP_REDUCE_WITH_OTHER_COMMONS **
[1.709] ** STEP_ENSURE_GEOHASH_RESULTS_ARE_INCLUDED_IF_CENTER_IS_GIVEN **
[1.719] ** STEP_AUTOCOMPLETE **
[1.731] Autocompleting provens
[1.830] Found tokens to autocomplete set()
[1.846] ** STEP_CHECK_BUCKET_FULL **
[1.856] ** STEP_CHECK_CREAM **
[1.866] ** STEP_FUZZY **
[1.881] Fuzzy on. Trying with [<Token brogli>, <Token 13100>, <Token aix>].
[1.903] Going fuzzy with brogli
[3.892] Found fuzzy candidates []
[3.914] Going fuzzy with aix
[4.965] Found fuzzy candidates []
[4.988] Fuzzy on. Trying with [<Token brogli>, <Token 13100>, <Token aix>].
[5.007] Going fuzzy with brogli
[6.687] Found fuzzy candidates []
[6.708] Going fuzzy with aix
[7.697] Found fuzzy candidates []
[7.715] ** STEP_EXTEND_RESULTS_REDUCING_TOKENS **
[7.728] Bucket dry. Trying to remove some tokens.
[7.746] Adding to bucket with keys ['w|brogli', 'w|aix']
[8.057] 2 ids in bucket so far
[8.075] Adding to bucket with keys ['w|brogli', 'w|13100']
[8.301] 2 ids in bucket so far
[8.318] Adding to bucket with keys ['w|13100', 'w|aix']
[9.916] 100 ids in bucket so far
[9.934] Computing results
[83.96] Done computing results
25 Rue Louis de Broglie 13290 Aix-en-Provence (ADRNIVX_0000000264620975 | importance: 0.0248/0.1, str_distance: 0.8723/1.0)
25 Rue de la Couronne 13100 Aix-en-Provence (ADRNIVX_0000000264584987 | importance: 0.0254/0.1, str_distance: 0.6538/1.0)
25 Rue d'Italie 13100 Aix-en-Provence (ADRNIVX_0000000264586580 | importance: 0.0278/0.1, str_distance: 0.6327/1.0)
25 Rue de la Verrerie 13100 Aix-en-Provence (ADRNIVX_0000000264582737 | importance: 0.0263/0.1, str_distance: 0.6226/1.0)
25 Rue de Cuques 13100 Aix-en-Provence (ADRNIVX_0000000312953568 | importance: 0.0271/0.1, str_distance: 0.62/1.0)
25 Rue de la Paix 13100 Aix-en-Provence (ADRNIVX_0000000264583091 | importance: 0.0271/0.1, str_distance: 0.6078/1.0)
25 Rue de l'Opéra 13100 Aix-en-Provence (ADRNIVX_0000000264584473 | importance: 0.0261/0.1, str_distance: 0.6078/1.0)
25 Rue Portalis 13100 Aix-en-Provence (ADRNIVX_0000000264581721 | importance: 0.0259/0.1, str_distance: 0.6/1.0)
25 Rue Lieutaud 13100 Aix-en-Provence (ADRNIVX_0000000264582642 | importance: 0.0258/0.1, str_distance: 0.6/1.0)
25 Rue des Cordeliers 13100 Aix-en-Provence (ADRNIVX_0000000264582531 | importance: 0.0288/0.1, str_distance: 0.5926/1.0)
84.6 ms / 10 results

**Comment by cquest** [comment]
Le CP indiqué est incorrect (13100 au lieu de 13290).
Un filtre sur citycode est utile dans ces cas...

Test OK sur version addok 1.0.0a avec trigrams si on filtre sur citycode:

http://ban.addok.xyz/search?q=25%20rue%20Louis%20de%20Broglie%20zac%20de%20la%20duranne%2013100%20Aix-en-Provence&citycode=13001

**Comment by yohanboniface** [comment]
If we bump COMMON_THRESHOLD to 15000, this will be fixed by #275.
Now let's make some test to see if we want to change this setting.

**Comment by cquest** [comment]
Plus de problème avec rc2

```
> EXPLAIN 25 rue Louis de Broglie - ZAC de la Duranne 13100 Aix-en-Provence
[5.718] Taken tokens: [<Token laduran>, <Token zak>, <Token brogli>, <Token 13100>, <Token duran>, <Token provens>, <Token aix>]
[5.734] Common tokens: [<Token loui>, <Token en>, <Token ru>, <Token la>, <Token de>, <Token de>]
[5.741] Housenumbers token: [<Token 25>]
[5.745] Not found tokens: []
[5.751] Filters: []
[5.759] ** ONLY_COMMONS_BUT_GEOHASH_TRY_AUTOCOMPLETE_COLLECTOR **
[5.766] ** NO_TOKENS_BUT_HOUSENUMBERS_AND_GEOHASH **
[5.771] ** NO_AVAILABLE_TOKENS_ABORT **
[5.776] ** ONLY_COMMONS **
[5.783] ** NO_MEANINGFUL_BUT_COMMON_TRY_AUTOCOMPLETE_COLLECTOR **
[5.788] ** ONLY_COMMONS_TRY_AUTOCOMPLETE_COLLECTOR **
[5.793] ** BUCKET_WITH_MEANINGFUL **
[5.808] New bucket with keys ['w|laduran', 'w|zak', 'w|brogli', 'w|13100', 'w|duran', 'w|provens', 'w|aix'] and limit 10
[6.112] 0 ids in bucket so far
[6.125] ** REDUCE_WITH_OTHER_COMMONS **
[6.142] ** ENSURE_GEOHASH_RESULTS_ARE_INCLUDED_IF_CENTER_IS_GIVEN **
[6.149] ** FUZZY_COLLECTOR **
[6.169] Fuzzy on. Trying with [<Token laduran>, <Token zak>, <Token brogli>, <Token 13100>, <Token duran>, <Token provens>, <Token aix>].
[6.191] Going fuzzy with laduran and ['w|zak', 'w|brogli', 'w|13100', 'w|duran', 'w|provens', 'w|aix', 'w|loui', 'w|en', 'w|ru', 'w|la', 'w|de', 'w|de']
[9.341] Going fuzzy with provens and ['w|laduran', 'w|zak', 'w|brogli', 'w|13100', 'w|duran', 'w|aix', 'w|loui', 'w|en', 'w|ru', 'w|la', 'w|de', 'w|de']
[13.15] Going fuzzy with brogli and ['w|laduran', 'w|zak', 'w|13100', 'w|duran', 'w|provens', 'w|aix', 'w|loui', 'w|en', 'w|ru', 'w|la', 'w|de', 'w|de']
[15.12] Going fuzzy with duran and ['w|laduran', 'w|zak', 'w|brogli', 'w|13100', 'w|provens', 'w|aix', 'w|loui', 'w|en', 'w|ru', 'w|la', 'w|de', 'w|de']
[16.81] Going fuzzy with zak and ['w|laduran', 'w|brogli', 'w|13100', 'w|duran', 'w|provens', 'w|aix', 'w|loui', 'w|en', 'w|ru', 'w|la', 'w|de', 'w|de']
[17.95] Going fuzzy with aix and ['w|laduran', 'w|zak', 'w|brogli', 'w|13100', 'w|duran', 'w|provens', 'w|loui', 'w|en', 'w|ru', 'w|la', 'w|de', 'w|de']
[19.10] Fuzzy on. Trying with [<Token laduran>, <Token provens>, <Token brogli>, <Token 13100>, <Token duran>, <Token zak>, <Token aix>].
[19.12] Going fuzzy with laduran and ['w|zak', 'w|brogli', 'w|13100', 'w|duran', 'w|provens', 'w|aix']
[21.24] Going fuzzy with provens and ['w|laduran', 'w|zak', 'w|brogli', 'w|13100', 'w|duran', 'w|aix']
[23.49] Going fuzzy with brogli and ['w|laduran', 'w|zak', 'w|13100', 'w|duran', 'w|provens', 'w|aix']
[25.45] Going fuzzy with duran and ['w|laduran', 'w|zak', 'w|brogli', 'w|13100', 'w|provens', 'w|aix']
[27.23] Going fuzzy with zak and ['w|laduran', 'w|brogli', 'w|13100', 'w|duran', 'w|provens', 'w|aix']
[28.40] Going fuzzy with aix and ['w|laduran', 'w|zak', 'w|brogli', 'w|13100', 'w|duran', 'w|provens']
[29.68] ** AUTOCOMPLETE_MEANINGFUL_COLLECTOR **
[29.70] Autocompleting provens
[29.91] Found tokens to autocomplete set()
[29.92] ** EXTEND_RESULTS_EXTRAPOLING_RELATIONS **
[38.17] Adding to bucket with keys ['w|aix', 'w|provens', 'w|duran']
[39.38] 3 ids in bucket so far
[39.39] Adding to bucket with keys ['w|aix', 'w|provens', 'w|zak']
[39.65] 9 ids in bucket so far
[39.66] Adding to bucket with keys ['w|aix', 'w|provens', 'w|brogli']
[39.87] 10 ids in bucket so far
[39.88] Adding to bucket with keys ['w|provens', 'w|duran', 'w|13100']
[40.32] 12 ids in bucket so far
[40.35] Adding to bucket with keys ['w|duran', 'w|laduran']
[40.65] 20 ids in bucket so far
[40.66] No relation extrapolated.
[40.67] ** EXTEND_RESULTS_REDUCING_TOKENS **
[40.68] Computing results
[40.70] Done getting results data
[43.34] Done computing results
Rue Louis De Broglie 13080 Aix-en-Provence (oZl4B | importance: 0.0567/0.1, str_distance: 0.6124/1.0)
Rue de la Petite Duranne 13080 Aix-en-Provence (PjVKn | importance: 0.0241/0.1, str_distance: 0.5323/1.0)

```
